lib/Configuration.php

```diff
@@ -80,31 +82,40 @@ class Configuration
 
     /**
      * Debug file location (log to STDOUT by default)
-nnn     *
+     *
      * @var string
      */
     protected $tempFolderPath;
 
+    /**
+     * @var ConfigurationOptions|null
+     */
+    protected $configurationOptions;
+
     /**
      * Constructor
+     * @param ConfigurationOptions|null $configurationOptions
+     * @param string|null $spapiEndpoint Note: This will override the ConfigurationOptions and ENV spapiEndpoint
```

Comment:
I agree, but for backwards compatibility sake, I figured this would allow people that might be initializing the Configuration will a host, it help with backwards compatibility.

```
$config = new Configuration(null, "HOST");
```

Reply:
Good point -- we can keep it till the major version bump.